dart/lib/src/transport/http_transport.dart

```diff
@@ -54,6 +70,8 @@ class HttpTransport implements Transport {
       body: body,
     );
 
+    _updateRetryAfterLimits(response);
```

Comment:
does `httpClient.post` throw? if so, this should be under a finally block

Reply:
Yep, at least in Dart:io environments it can throw `SocketException`s

Reply:
Not sure if i understand. If we throw, we also don't have a response to call `_updateRetryAfterLimits` with, or am I missing something else here?

Reply:
@denrase it depends on when/how it throws.

Some libs throw an error with 4xx and 5xx but you can still read the response.
Some libs (maybe the use case for Dart HTTP), do not throw in such cases, so the current implementation is just fine.